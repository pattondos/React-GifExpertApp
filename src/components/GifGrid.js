import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {loading && <p className="animate__animated animate__flash">Loading</p>}

            <div className="card-grid">
                <ol>
                    {images.map(gif => (
                        <GifGridItem
                            key={gif.id}
                            {...gif}
                        />
                    ))
                    }
                </ol>
            </div>
        </>
    )
}

export default GifGrid;
