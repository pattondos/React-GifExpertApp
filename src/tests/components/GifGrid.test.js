import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas para el <GifGrid />', () => {

    const category = 'Hola';

    test('Should be use the default options when options is defined ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should be use the default options when options is not defined', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cosa',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


});
