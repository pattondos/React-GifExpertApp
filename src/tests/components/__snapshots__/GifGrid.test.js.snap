// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el <GifGrid /> Should be use the default options when options is defined  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas para el <GifGrid /> Should be use the default options when options is not defined 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cosa"
        url="https://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;
